Add answer links to the knowledge check section of Ruby/Nested Collections (#23017)
TheOdinProject/curriculum: ruby_programming/basic_ruby/nested_collections.md

#=> nil
~~~

> <span id='dig-method-knowledge-check'>If you want a nil value returned when trying to access an index of a nonexistent nested element, you can use the `#dig` method.<span> This method can also be used when accessing a nonexistent index inside of an existing nested element.

Reviewer: ```suggestion
<span id='dig-method'>If you want a nil value returned when trying to access an index of a nonexistent nested element, you can use the `#dig` method.<span> This method can also be used when accessing a nonexistent index inside of an existing nested element.
```

### Knowledge Check
This section contains questions for you to check your understanding of this lesson. If you're having trouble answering the questions below on your own, review the material above to find the answer.

- <a class='knowledge-check-link' href='#nested-arrays'>What is a nested array? What data is useful to store in a nested array?</a>
- <a class='knowledge-check-link' href='#nested-hashes'>What is a nested hash? What data is useful to store in a nested hash?</a>
- <a class='knowledge-check-link' href='#accessing-elements'>How do you access elements in a nested array?</a>
- <a class='knowledge-check-link' href='#accessing-data'>How do you access data in a nested hash?</a>

> - <a class='knowledge-check-link' href='#dig-method-knowledge-check'>Why is the `#dig` method useful?<a>

Reviewer: ```suggestion
- <a class='knowledge-check-link' href='#dig-method'>Why is the `#dig` method useful?<a>
```

- <a class='knowledge-check-link' href='#adding-and-removing-elements'>How do you add elements to a nested array?</a>
- <a class='knowledge-check-link' href='#adding-and-removing-data'>How do you add data to a nested hash?</a>
- <a class='knowledge-check-link' href='#remove-elements-nested-array'>How do you delete elements from a nested array?</a>
- <a class='knowledge-check-link' href='#deleting-data-nested-hash'>How do you delete data in a nested hash?</a>
- <a class='knowledge-check-link' href='#create-immutable-nested-arrays'>How do you create a new nested array that is not mutable?</a>

> - <a class='knowledge-check-link' href='#iterating-over-nested-a-array'>How do you iterate over a nested array?</a>

Reviewer: ```suggestion
- <a class='knowledge-check-link' href='#iterating-over-a-nested-array'>How do you iterate over a nested array?</a>
```
